ironfish/src/wallet/wallet.ts

```diff
@@ -1526,6 +1533,28 @@ export class Wallet {
         }
       }
 
+      if (identity) {
```

Comment:
do you think we could combine this with the `if (secret)` block, above? if there is a secret passed in then we can derive the identity

Reply:
I wrote this so that they both wouldn't be defined at the same time (only 1 or the other). treating each account type hardware vs signer separately. 

does it still make sense to combine these?

Reply:
i think it makes sense to combine them because `identity` will always be defined if `secret` is defined. the `if (secret)` case becomes the `if (identity)` case if we derive the `identity` from the secret